src/plugins/workspace/public/components/workspace_form/utils.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DEFAULT_CHECKED_FEATURES_IDS } from '../../../common/constants';
+
+import { WorkspaceFeature, WorkspaceFeatureGroup, WorkspaceFormErrors } from './types';
+
+export const isWorkspaceFeatureGroup = (
+  featureOrGroup: WorkspaceFeature | WorkspaceFeatureGroup
+): featureOrGroup is WorkspaceFeatureGroup => 'features' in featureOrGroup;
+
+export const isDefaultCheckedFeatureId = (id: string) => {
+  return DEFAULT_CHECKED_FEATURES_IDS.indexOf(id) > -1;
+};
+
+export const appendDefaultFeatureIds = (ids: string[]) => {
+  // concat default checked ids and unique the result
+  return Array.from(new Set(ids.concat(DEFAULT_CHECKED_FEATURES_IDS)));
+};
+
+export const isValidNameOrDescription = (input?: string) => {
```

Comment:
Is it possible to add a comment or inline documentation explaining the purpose and restrictions of the regular expression used in the isValidNameOrDescription function. Thanks

Reply:
This is for align with the UI.
![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/4034161/eb6a6f00-fa2c-46c6-9c5f-7ae2daa61519) Will add a comment to explaining it.